ols/utils/auth_dependency.py

```diff
@@ -206,7 +238,7 @@ async def __call__(self, request: Request) -> tuple[str, str]:
                 status_code=403, detail="Forbidden: Invalid or expired token"
             )
         if user_info.user.username == "kube:admin":
-            user_info.user.uid = DEFAULT_KUBEADMIN_UID
+            user_info.user.uid = get_cluster_id()
```

Comment:
the cluster id isn't going to change between user queries, so we should just fetch it once during OLS startup and then use that value every time we get a request from kubeadmin.

this will also help avoid anther issue you're not handling here, which is what happens if getting the cluster id fails?  you log the errors in the helper, but ultimately return `None` to the caller.

if you fetch it during startup you can retry + refuse to start until we successfully get the cluster id.

it also removes the extra/repeated api calls.